### ISSUE TITLE
New snippet overwrites the older one

### ISSUE THREAD

**Issue by sorokinpsi** [paulscherrerinstitute/scilog#195]
In this logbook:
https://scilog.psi.ch/logbooks/63d55df141019825a5a7e315/dashboard

currently, snippet 9 from 3:40 pm is exactly the same as snippet 16 from 10:36 pm. I'm pretty sure that I posted that text only once, that is at 10:36 pm. I don't know if it is somehow related, but I might have edited snippet 9 right before adding snippet 16.

**Comment by stephan271** [comment]
Is this entry meant for issue #193 ?

**Comment by sorokinpsi** [comment]
No, It's a separate issue. In issue #193 some snippets a re just not visible in the browser, but can be recovered in the pdf export. Here the snippet 9 was completely overwritten by snippet 16, and cannot be recovered even in the pdf.

**Comment by stephan271** [comment]
Okay, thanks

**Comment by stephan271** [comment]
Did the editing of snippet 9 and the adding of snippet 16 happen from the same session and the same PCone after another ? Or did you have 2 concurrent sessions open ?

**Comment by sorokinpsi** [comment]
Most likely, the editing and the adding happened from the same session, but there were several other 'idle' sessions opened on other beamline PCs. I need to say though, that everything have happened during a hectic beamtime, so I can't be 100% sure right now.

**Comment by stephan271** [comment]
Okay, thanks. So far we could not reproduce the effect here, but we can confirm that something went wrong in your case. Still need to find out what conditions are needed to trigger this problem...

**Comment by stephan271** [comment]
From what I can see on the database level is, that **snippet number 9** has been edited 41 times:

- 2023-02-07T14:40:43.905 <p>15:38 - the I0 measurement was started</p>
- 2023-02-07T14:40:46.323 <p>15:38 - the I0 measurement was started</p>
- 2023-02-07T14:40:51.086 <p>15:38 - the I0 measurement was started.&nbsp;</p>
- 2023-02-07T14:40:53.573 <p>15:38 - the I0 measurement was started. Gas pressure&nbsp;<p>
- ...
- 2023-02-07T20:57:17.360 <p>Sample detector distance on 28.01.23 - 7.7 cm</p>
- ...
- 2023-02-07T21:36:32.691 <p>The sample has been rotated 90 deg anticlockwise looking at the front of the sample. The x-ray&nbsp;</p>
- 2023-02-07T21:36:32.695 <p>The sample has been rotated 90 deg anticlockwise looking at the front of the sample. The x-ray&nbsp;</p>


**Snippet number 16** was created with

- 2023-02-07T21:36:33.592+00:00 : <p>The sample has been rotated 90 deg anticlockwise looking at the front of the sample. The x-ray illuminate the sample from the back.</p><p>&nbsp;</p>

So it looks as if 1 sec before creating snippet 16 snippet 9 was updated with a contents similar to what is posted to snippet 16 (the empty paragraph at the end is missing)

The easiest explanation for this would be that the sentence was first added to the edit window of snippet 9 (by accident, I assume) and then added as snippet 16.

Would this be a possible explanantion ?

**Comment by sorokinpsi** [comment]
I definitely did not confused the editing window of snippet 9 with a box for making new snippets 41 times in a row. To me it seems like the editing window of snippet 9 was open on another machine (in the experimental hutch) all the time, while new snippets were added from the PC in the control room. Perhaps, this simultaneous editing from one PSI account but two different machines lead to older snippet being overwritten.

In general, I did minor corrections to the snippets (like typos), but I barely changed the whole text of the snippet. I hope it helps.

**Comment by stephan271** [comment]
Sure it helps, thanks for clarifications, we are still investigating what the reason could be. In fact, we can see, that all entries were made from a single account

**Comment by stephan271** [comment]
And just for clarification: the confusion editing/adding only new to have happened **once**, not 41 times to explain the observed effect. But it could well be an issue with concurrent sessions as well, we will find out ;-)

**Comment by stephan271** [comment]
We have fixed #129 , which we think might be the cause of the issues reported here. Your problem would be follow-up problem of the initial image editing problem. So we will close this issue for now, since the bugfixes have been deployed to the prod system. However if you encounter the problem again please reopen the issue. Thanks for your feedback in any case !
